From 3c075fca2edccdefb2b91de61090f8f1219c54a7 Mon Sep 17 00:00:00 2001
From: Slava Monich <slava.monich@jolla.com>
Date: Tue, 7 Apr 2015 16:13:07 +0300
Subject: [PATCH 20/22] ipv6pd: Check timer_hash for NULL prior to destroying
 it

GLib doesn't like NULL pointers.

cleanup() may be invoked more than once after successful setup,
first from dhcpv6_callback() and then from __connman_ipv6pd_cleanup().

(cherry picked from commit 2425d95e203affa769f7298803c79c0f11512d97)
---
 src/ipv6pd.c | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/src/ipv6pd.c b/src/ipv6pd.c
index 5ecda38..0d221f0 100644
--- a/src/ipv6pd.c
+++ b/src/ipv6pd.c
@@ -165,8 +165,10 @@ static void cleanup(void)
 		timer_uplink = 0;
 	}
 
-	g_hash_table_destroy(timer_hash);
-	timer_hash = NULL;
+	if (timer_hash) {
+		g_hash_table_destroy(timer_hash);
+		timer_hash = NULL;
+	}
 
 	if (prefixes) {
 		g_slist_free_full(prefixes, g_free);
-- 
1.9.1

